Server/ConcordiaCurriculumManager/Services/CourseService.cs

```diff
@@ -12,7 +12,7 @@ public interface ICourseService
     public IEnumerable<CourseCareerDTO> GetAllCourseCareers();
     public IEnumerable<CourseComponentDTO> GetAllCourseComponents();
     public Task<IEnumerable<string>> GetAllCourseSubjects();
-    public Task<CourseCreationDossier> InitiateCourseCreation(CourseCreationInitiationDTO initiation, User user);
+    public Task<CourseCreationRequest> InitiateCourseCreation(CourseCreationInitiationDTO initiation, User user, Guid dossierId);
```

Comment:
As with the controller, DossierId should be part of CourseCreationInitiationDTO.